packages/web-workers/src/webpack-parts/loader.ts

```diff
@@ -89,13 +89,11 @@ export function pitch(
     );
   }
 
+  const {filename, chunkFilename} = compiler.options.output!;
+
   const workerOptions = {
-    filename: addWorkerSubExtension(
-      compiler.options.output!.filename as string,
-    ),
-    chunkFilename: addWorkerSubExtension(
-      compiler.options.output!.chunkFilename!,
-    ),
+    filename: addWorkerSubExtension((chunkFilename || filename) as string),
```

Comment:
I don't think this is the fix we want. It works in the specific case you have, but someone who either doesn't have `[name]` in the `chunkFilename`, or who actually wants to use the `filename` as the base for their worker filename, will not be able to with this approach. `chunkFilename` might also have some substitutions that don't work as a `filename`.

After giving it a bit more thought, I think that the best approach would be to keep using `filename`/ `chunkName` as we were before, but for `addWorkerSubExtension`, we should change the logic to this:

```ts
return file.includes('[name]') ? file.replace(/\.([a-z]+)$/i, '.worker.$1')  : file.replace(/\.([a-z]+)$/i, '.[name].worker.$1');
```

Reply:
This logic just attempts to append `worker` to the worker file name. It's not possible to add other things like contenthash just for worker file but not the main entry which is what we want for Argo Admin. What option do we have if we want that use case?

Reply:
Could we add a custom `workerFileName` option to this plugin that the user can pass in? It can default to `compiler.options.output!.filename` like before.

Reply:
@vividviolet It's not possible to add unsupported key to Webpack config (ex: workerFilename) because Webpack validates all the keys and throw exception.

Reply:
Can we not pass the login directly to this plugin when initiating it?

Reply:
If we need a specific filename for the worker created for argo-admin, I agree with Trish that we should have an option on the loader for it. The default should just do what is usually right, which is to include the hash and a marker that the file is a worker (hence the `.worker.` path part).

Reply:
That works 👍